src/content/subgames.cpp

```diff
@@ -288,8 +306,27 @@ std::vector<WorldSpec> getAvailableWorlds()
 	return worlds;
 }
 
-bool loadGameConfAndInitWorld(const std::string &path, const SubgameSpec &gamespec)
+bool loadGameConfAndInitWorld(const std::string &path, const std::string &name,
+		const SubgameSpec &gamespec, bool mustCreateNew, std::string *error)
 {
+	std::string final_path = path;
+
+	// If we're creating a new world, ensure that the path isn't already taken
+	if (mustCreateNew) {
+		int counter = 1;
+		while (fs::PathExists(final_path) && counter < MAX_WORLD_NAMES) {
+			final_path = path + "_" + std::to_string(counter);
+			counter++;
+		}
+
+		if (fs::PathExists(final_path)) {
+			if (error != nullptr) {
+				*error = "Too many similar filenames";
```

Comment:
Wouldn't this be an ideal use-case for exceptions?

Reply:
I'm always a little wary of throwing exceptions in a gamedev context, but since this isn't a hotspot it should be fine. Do you know any existing exceptions that would fit the bill here, or would you prefer that I make a new one?

Reply:
A generic exception (`BaseException`) likely suffices here. Yet I'll leave the decision whether to use exceptions to you.
Oh and `InvalidFilenameException` is currently unused and could be removed...

Reply:
I thought this was a good suggestion at first, but after writing most of the necessary code I found that it makes the error checking in `l_mainmenu.cpp` kinda messy as a result... we could make it work by making this function return void and throwing an exception for the other failure state as well, but I'd like your opinion on that first.

Reply:
It would be simple as
```C++
void loadGameConfAndInitWorld()
{
	throw BaseException("Some error");
}

try {
	loadGameConfAndInitWorld(path, name, games[gameidx], true))
	lua_pushnil(L);
} catch (BaseException &e) {
	lua_pushstring(L, ("Failed to initialize world: " + e.what()).c_str());
}
```

Reply:
Gotcha, I'll go ahead and replace both then :)